### PR TITLE
Fix crashes related to disabling the depth sensors at launch.

### PR DESCRIPTION
**Pull request by J-Rojas**

### CODE REVIEW

**Comment by J-Rojas** [comment]
Fix for issue #429

File: realsense2_camera/src/base_realsense_node.cpp [hunk 1]
```diff
@@ -87,7 +87,9 @@ void BaseRealSenseNode::publishTopics()
     setupDevice();
     setupPublishers();
     setupStreams();
-    publishStaticTransforms();
+    if (_enable[DEPTH]) {
+      publishStaticTransforms();
```
Comment (shuntaraw) [inline realsense2_camera/src/base_realsense_node.cpp:91]: We want tf for enabled streams to be published, such as color_frame to camera_link even if depth is disabled.

Comment (J-Rojas) [reply realsense2_camera/src/base_realsense_node.cpp:91]: Okay, the publishStaticTransforms() function would have to be refactored so the other tf frames are independent of the depth frame_id. They are currently coupled to the extrinsics of the depth camera, which IIRC, are not available if the depth camera is disabled, causing a crash.

**Comment by doronhi** [comment]
Added ability in PR#601. Thanks for the code tips and insights.

**Pull request closed without merging**